util/giveRoleAfterDelay

```diff
@@ -0,0 +1,37 @@
+{{/*
+	This command gives a role *after* delay
+	Usage: "-timedrole @user [time] role_name" [time] can be 10d, 10m, 10s etc.
+ 
+	Recommended trigger: timedrole
+	Trigger type: Command
+ 
+	
+*/}}
+{{/*ACTUAL CODE, DO NOT TOUCH*/}}
+{{ if not .ExecData }}
+{{ $args := parseArgs 3 "Wrong usage: `-timedrole (@user) (time) (role name)`" 
+ (carg "userid" "member to give role" )
+ (carg "duration" "time till role" )
+ (carg "string" "role name" ) }}
+{{ $search := slice .CmdArgs 2 | joinStr "" | lower }} 
+ {{ $role := 0 }} {{/* Role variable */}}
+ {{/* Search for roles */}}
+ {{- range .Guild.Roles -}}
+ {{ if not $role }}
+ {{ $rName := lower .Name }}
+ {{ if or (eq $search $rName) (in $rName $search) }}
+ {{ $role = . }}
+ {{ end }}
+ {{ end }}
+ {{- end -}}
+{{ if $role }}
+{{ $a := (toDuration ($args.Get 1)).Seconds }}
+{{ if not (hasRoleName ($args.Get 2)) }}
+{{ execCC .CCID nil $a (sdict "user" ($args.Get 0) "role" ($args.Get 2)) }}
```

Comment:
this won't work properly with fuzzy matching. Say user provides `foo` as their input role and the role that matches is `foobar`. `foo` is not the role name, `foobar` is, but you're using `foo`. Use role ID instead.